### PR TITLE
Drone

### PR DESCRIPTION
**Pull request by klemens-morgenstern**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,12 +6,13 @@
 cmake_minimum_required(VERSION 3.5...3.16)
 
 project(boost_process VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
+find_package(Boost REQUIRED iostreams program_options filesystem system thread)
```
[the remark this answers is not in the document]
Comment (pdimov) [reply CMakeLists.txt:9]: Confirmed. The Boost CMake build is broken now, as a result of this change. Please fix.

File: include/boost/process/environment.hpp [hunk 2]
```diff
@@ -263,7 +263,9 @@ class basic_environment_impl : public Implementation<Char>
         auto st1 = key + ::boost::process::detail::equal_sign<Char>();
         while (*p != nullptr)
         {
-            if (std::equal(st1.begin(), st1.end(), *p))
+            const int len = std::char_traits<Char>::length(*p);
```
Comment (kou) [inline include/boost/process/environment.hpp:266]: Can we use `iterator::difference_type` here something like `const auto len = static_cast<iterator::difference_type>(std::char_traits<Char>::length(*p))`?
This causes an warning with Visual C++:

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/44489657/job/xh0s898nasxat5fn#L1515

```text
C:\Miniconda37-x64\envs\arrow\Library\include\boost/process/environment.hpp(266): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
C:\Miniconda37-x64\envs\arrow\Library\include\boost/process/environment.hpp(266): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
```

File: include/boost/process/environment.hpp [hunk 3]
```diff
@@ -288,7 +292,9 @@ class basic_environment_impl : public Implementation<Char>
         auto st1 = st + ::boost::process::detail::equal_sign<Char>();
         while (*p != nullptr)
         {
-            if (std::equal(st1.begin(), st1.end(), *p))
+            const int len = std::char_traits<Char>::length(*p);
```
Comment (kou) [inline include/boost/process/environment.hpp:295]: Ditto.

File: include/boost/process/environment.hpp [hunk 4]
```diff
@@ -275,7 +277,9 @@ class basic_environment_impl : public Implementation<Char>
         auto st1 = key + ::boost::process::detail::equal_sign<Char>();
         while (*p != nullptr)
         {
-            if (std::equal(st1.begin(), st1.end(), *p))
+            const int len = std::char_traits<Char>::length(*p);
```
Comment (kou) [inline include/boost/process/environment.hpp:280]: Ditto.

**Pull request merged**